lib/translator/translation.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "translation.h"
+
+namespace translator {
+
+BeginResponse Translation::Begin(absl::Span<const uint8_t> scsi_cmd,
+                                 scsi_defs::LunAddress lun) {
+  BeginResponse response = {};
+  response.status = ApiStatus::kSuccess;
+  if (pipeline_status_ != StatusCode::kUninitialized) {
+    DebugLog("Invalid use of API: Begin called before complete or abort");
+    response.status = ApiStatus::kFailure;
+    return response;
+  }
+
+  // Verify buffer is large enough to contain opcode (one byte)
+  if (scsi_cmd.empty()) {
+    DebugLog("Empty SCSI Buffer");
+    pipeline_status_ = StatusCode::kFailure;
+    return response;
+  }
+
+  pipeline_status_ = StatusCode::kSuccess;
+  scsi_cmd_ = scsi_cmd;
+
+  scsi_defs::OpCode opc = static_cast<scsi_defs::OpCode>(scsi_cmd[0]);
```

Comment:
Wondering if it's worth having a `raw_scsi = scsi_cmd_.subspan(1)` or something so we don't have to write `.subspan(1)` in every case.

Reply:
I'll add this in, it seems like it will most likely make everyone's life easier when testing